client/src/www/views/root_view/root_header/index.ts

```diff
@@ -0,0 +1,97 @@
+/*
+  Copyright 2024 The Outline Authors
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+       http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+import {LitElement, html, css} from 'lit';
+import {customElement, property} from 'lit/decorators.js';
+import {classMap} from 'lit/directives/class-map.js';
+
+@customElement('root-header')
+export class RootHeader extends LitElement {
+  @property({type: String}) title = '';
+  @property({type: Boolean}) showBackButton = false;
+  @property({type: Boolean}) showAddButton = false;
+
+  static styles = css`
+    header {
+      align-items: center;
+      justify-content: space-between;
+      background-color: var(--outline-dark-primary);
+      display: flex;
+      height: 64px;
+      padding: 0 16px;
+    }
+
+    h1 {
+      color: var(--outline-white);
+      font-family: 'Jigsaw Sans', 'Roboto', sans-serif;
+      font-size: 24px;
+      font-weight: 500;
+      margin: 0;
+    }
+
+    md-icon {
+      color: var(--outline-white);
+    }
+
+    .hidden {
+      visibility: hidden;
+    }
+  `;
+
+  render() {
+    return html`<header>
+      ${this.showBackButton
+        ? html`<md-icon-button @click=${this.handleReturnToServers}>
+            <md-icon>arrow_back</md-icon>
+          </md-icon-button>`
+        : html`<md-icon-button @click=${this.handleMenu}>
+            <md-icon>menu</md-icon>
+          </md-icon-button>`}
+      <h1>${this.title || 'Outline'}</h1>
+      <md-icon-button
+        class=${classMap({hidden: !this.showAddButton})}
+        @click=${this.handleAdd}
+      >
+        <md-icon>add</md-icon>
+      </md-icon-button>
+    </header>`;
+  }
+
+  handleAdd() {
+    this.dispatchEvent(
+      new CustomEvent('ShowAddServerDialog', {
+        bubbles: true,
+        composed: true,
+      })
+    );
+  }
+
+  handleMenu() {
+    this.dispatchEvent(
+      new CustomEvent('ShowNavigation', {
+        bubbles: true,
+        composed: true,
+      })
+    );
+  }
+
+  handleReturnToServers() {
```

Comment:
Could we rename this to `handleBack()` and send a `ChangePage` for `page: 'home'`? Or leverage `DEFAULT_PAGE` somehow as a shared constant?

The homepage just happens to be the server page, but the header shouldn't need to know that detail.

Reply:
Even better, as it's preferred to `handle...` type naming.